webpack/components/AnsibleRolesAndVariables/AnsibleRolesAndVariables.js

```diff
@@ -0,0 +1,178 @@
+
+  const [isChecked, setIsChecked] = useState(false);
+  const [selectedRowsCount, setSelectRowsCount] = useState(0);
+
+  const [page, setPage] = useState();
+  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE);
+  const [paginatedRows, setPaginatedRows] = useState(rows.slice(0, perPage));
+
+  const onSelect = (event, isSelected, rowId, row) => {
+    const selectableRowLength = rows.filter(
+      tempRow => tempRow.parent === undefined
+    ).length;
+    let rowsCount = selectedRowsCount;
+    const tempRows = rows.map(tempRow => {
+      if (rowId === -1) {
+        tempRow.selected = isSelected;
+        setIsChecked(isSelected);
+        setSelectRowsCount(isSelected ? selectableRowLength : 0);
+      } else if (tempRow.id === row.id) {
+        tempRow.selected = isSelected;
+        rowsCount = isSelected ? selectedRowsCount + 1 : selectedRowsCount - 1;
+        if (rowsCount > 0 && rowsCount === selectableRowLength) {
+          setIsChecked(true);
+        } else {
+          setIsChecked(false);
+        }
+        setSelectRowsCount(rowsCount);
+      }
+      return tempRow;
+    });
+    setRows(tempRows);
+  };
+
+  const SelectAll = checked => {
+    onSelect(null, checked, -1);
+  };
+
+  const handleSetPage = (event, newPage) => {
+    const startIdx = (newPage - 1) * perPage;
+    const endIdx =
+      rows.length < newPage * perPage ? rows.length : newPage * perPage;
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+  };
+
+  const handlePerPageSelect = (
+    event,
+    newPerPage,
+    newPage,
+    startIdx,
+    endIdx
+  ) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+  };
+
+  const renderPagination = (variant = 'top') => (
+    <Pagination
+      isCompact
+      itemCount={rows.length}
+      page={page}
+      perPage={perPage}
+      defaultToFullPage
+      onSetPage={handleSetPage}
+      onPerPageSelect={handlePerPageSelect}
+      perPageOptions={[
+        { title: '3', value: 3 },
+        { title: '5', value: 5 },
+        { title: '10', value: 10 },
+      ]}
+      titles={{
+        paginationTitle: `${variant} pagination`,
+      }}
+    />
+  );
+
+  const renderSelectAll = () => (
+    <Toolbar>
+      <ToolbarContent>
+        <ToolbarGroup variant="icon-button-group">
+          <ToolbarItem>
+            <Checkbox
+              isChecked={isChecked}
+              onChange={SelectAll}
+              aria-label="select all checkbox"
+              id="select-all"
+              name="select-all"
+              label={isChecked ? 'Deselect all' : 'Select all'}
+            />
+          </ToolbarItem>
+        </ToolbarGroup>
+      </ToolbarContent>
+    </Toolbar>
+  );
+
+  const renderSubmitAndCancel = () => (
+    <div>
+      <br />
+      <br />
+      <Button variant="primary" onClick={() => onSubmit(rows, proxy)}>
+        Submit
+      </Button>
+      &nbsp;
```

Comment:
What do you think about using [css](https://github.com/theforeman/foreman_ansible/commit/b723d344e568af5afbde9e1b217194814f61cdc9) rather than adding nodes into page to achieve the desired look?

Reply:
done